src/memberPublicKey.spec.ts

```diff
@@ -0,0 +1,211 @@
+      expect(dbResult?.publicKey.toString('base64')).toStrictEqual(publicKey.toString('base64'));
+      expect(mockLogging.logs).toContainEqual(
+        partialPinoLog('info', 'Member public key created', {
+          id: memberPublicKey.result.id,
+        }),
+      );
+    });
+
+    test('Malformed public key should be refused', async () => {
+      const memberPublicKey = await createMemberPublicKey(
+        MEMBER_MONGO_ID,
+        {
+          publicKey: Buffer.from('invalid public key').toString('base64'),
+          oid: TEST_OID,
+        },
+        serviceOptions,
+      );
+
+      requireFailureResult(memberPublicKey);
+      expect(memberPublicKey.reason).toBe(MemberPublicKeyProblemType.MALFORMED_PUBLIC_KEY);
+    });
+
+    test('Record creation errors should be propagated', async () => {
+      await connection.close();
+
+      const error = await getPromiseRejection(
+        async () =>
+          createMemberPublicKey(
+            MEMBER_MONGO_ID,
+            {
+              publicKey: publicKey.toString('base64'),
+              oid: TEST_OID,
+            },
+            serviceOptions,
+          ),
+        Error,
+      );
+
+      expect(error).toHaveProperty('name', 'MongoNotConnectedError');
+    });
+  });
+
+  describe('getMemberPublicKey', () => {
+    test('Existing id should return the corresponding data', async () => {
+      const memberPublicKey = await memberPublicKeyModel.create({
+        memberId: MEMBER_MONGO_ID,
+        oid: TEST_OID,
+        publicKey,
+      });
+
+      const result = await getMemberPublicKey(
+        MEMBER_MONGO_ID,
+        memberPublicKey._id.toString(),
+        serviceOptions,
+      );
+
+      requireSuccessfulResult(result);
+      expect(result.result).toMatchObject({
+        oid: TEST_OID,
+        publicKey: publicKey.toString('base64'),
+      });
+    });
+
+    test('Non existent id should return non existing error', async () => {
+      const invalidPublicKeyId = '111111111111111111111111';
+      await memberPublicKeyModel.create({
+        memberId: MEMBER_MONGO_ID,
+        oid: TEST_OID,
+        publicKey,
+      });
+
+      const result = await getMemberPublicKey(MEMBER_MONGO_ID, invalidPublicKeyId, serviceOptions);
+
+      requireFailureResult(result);
+      expect(result.reason).toBe(MemberPublicKeyProblemType.PUBLIC_KEY_NOT_FOUND);
+    });
+
+    test('Non existent member id should return non existing error', async () => {
+      const invalidMemberKeyId = '111111111111111111111111';
+      await memberPublicKeyModel.create({
+        memberId: MEMBER_MONGO_ID,
+        oid: TEST_OID,
+        publicKey,
+      });
+
+      const result = await getMemberPublicKey(invalidMemberKeyId, MEMBER_MONGO_ID, serviceOptions);
+
+      requireFailureResult(result);
+      expect(result.reason).toBe(MemberPublicKeyProblemType.PUBLIC_KEY_NOT_FOUND);
+    });
+
+    test('Record Find errors should be propagated', async () => {
```

Comment:
We don't need this test here: We only need it in functions that could failed due to duplicated records (e.g., `createOrg()`).